hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaControllerManagement.java

```diff
@@ -177,19 +177,23 @@ public int getWeightConsideringDefault(final Action action) {
     }
 
     @Override
-    protected void onActionStatusUpdate(final Action.Status updatedActionStatus, final JpaAction action) {
+    protected void onActionStatusUpdate(final JpaActionStatus newActionStatus, final JpaAction action) {
+        Action.Status updatedActionStatus = newActionStatus.getStatus();
         switch (updatedActionStatus) {
             case ERROR: {
                 final JpaTarget target = (JpaTarget) action.getTarget();
                 target.setUpdateStatus(TargetUpdateStatus.ERROR);
+                action.setTimestamp(newActionStatus.getOccurredAt());
                 handleErrorOnAction(action, target);
                 break;
             }
             case FINISHED: {
+                action.setTimestamp(newActionStatus.getOccurredAt());
```

Comment:
!!! @ Here, probably in handleFinishedAndStoreInTargetStatus shall be checked:
* If the action timestamp is less than the action that has set the Target#installedDistributionSet - it shall not override
* this is the main goal of timestamp management

Reply:
yup,.. somehow removed from draft.. added it again. thanks